apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolOptions.cs

```diff
@@ -162,18 +163,39 @@ public TimeSpan Timeout
             set => _timeout = CheckPositiveTimeSpan(value);
         }
 
+        /// <summary>
+        /// The maximum number of sessions that will be created in a batch.
+        /// Sessions created are associated to the gRPC channel they are created on, so
+        /// all sessions created in a batch are associated to the same gRPC channel.
+        /// Batch size should be limited so as not to overload a given channel.
+        /// If the sessions needing to be created at any given time are more than this value
+        /// then multiple batches of this size or less will be created.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This value must be positive. This value should always be less than or equal to <see cref="MaximumConcurrentSessionCreates"/>
+        /// The default value is 5.
+        /// </para>
+        /// </remarks>
+        public int CreateSessionBatchMaximumSize
+        {
+            get => _createSessionBatchMaximumSize;
+            set => _createSessionBatchMaximumSize = GaxPreconditions.CheckArgumentRange(value, nameof(value), 1, MaximumConcurrentSessionCreates);
+        }
+
         /// <summary>
         /// The maximum number of session create operations allowed to occur simultaneously per session pool.
         /// Spanner has limits on the number of sessions that can be created concurrently without affecting performance.
         /// This value is not typically changed.
         /// </summary>
         /// <para>
-        /// This value must be positive. The default value is 10.
+        /// This value must be positive. This value should always be greater than or equal to <see cref="CreateSessionBatchMaximumSize"/>
+        /// The default value is 50.
         /// </para>
         public int MaximumConcurrentSessionCreates
         {
             get => _maximumConcurrentSessionCreates;
-            set => _maximumConcurrentSessionCreates = GaxPreconditions.CheckArgumentRange(value, nameof(value), 1, int.MaxValue);
+            set => _maximumConcurrentSessionCreates = GaxPreconditions.CheckArgumentRange(value, nameof(value), CreateSessionBatchMaximumSize, int.MaxValue);
```

Comment:
There's a problem with having these preconditions - it means you can end up with an object initializer which is valid in one ordering and invalid in another.

If MaximumConcurrentSessionCreates is less than CreateSessionBatchMaximumSize, it just means we'll always use a single batch - that's not unreasonable, really.

Reply:
True, and I've hated when that has happened to me.
And true as well that is no big deal to remove that validation.

Done.